### ISSUE TITLE
Test failures for where local and remote ports are wrong

### ISSUE THREAD

**Issue by paulkernfeld** [feross/simple-peer#63]
This failure happens for me in both wrtc and Firefox. This happens nondeterministically, maybe about 2/3 of the time with wrtc and 1/2 of the time with Firefox.

Sample wrtc test failure:
```
paul ~/repos/simple-peer master $ npm run test-node

> simple-peer@5.11.8 test-node /Users/paul/repos/simple-peer
> echo 'DISABLED' && tape test/*.js

DISABLED
TAP version 13
# get config
# detect WebRTC support
ok 1 builtin webrtc support
# signal event gets emitted
ok 2 got signal event
# data send/receive text
ok 3 (unnamed assert)
ok 4 (unnamed assert)
ok 5 peer1 is initiator
ok 6 peer2 is not initiator
not ok 7 should be equal
  ---
    operator: equal
    expected: '192.168.0.2'
    actual:   '10.100.1.6'
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:58:7)
  ...
not ok 8 should be equal
  ---
    operator: equal
    expected: 60221
    actual:   58969
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:59:7)
  ...
not ok 9 should be equal
  ---
    operator: equal
    expected: '192.168.0.2'
    actual:   '10.100.1.6'
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:61:7)
  ...
not ok 10 should be equal
  ---
    operator: equal
    expected: 62750
    actual:   58971
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:62:7)
  ...
ok 11 (unnamed assert)
ok 12 (unnamed assert)
ok 13 (unnamed assert)
ok 14 (unnamed assert)
ok 15 got correct message
ok 16 got correct message
ok 17 both peers closed
# sdpTransform function is called
ok 18 got a string as SDP
# get config
# data send/receive Uint8Array
ok 19 data is Buffer
ok 20 got correct message
ok 21 data is Buffer
ok 22 got correct message
ok 23 both peers closed
# data send/receive Buffer
ok 24 data is Buffer
ok 25 got correct message
ok 26 data is Buffer
ok 27 got correct message
ok 28 both peers closed
# data send/receive ArrayBuffer
ok 29 data is Buffer
ok 30 got correct message
ok 31 data is Buffer
ok 32 got correct message
ok 33 both peers closed
# get config
# duplex stream: send data before "connect" event
ok 34 got peer1 "finish"
ok 35 (unnamed assert)
ok 36 got correct message
ok 37 got peer1 "end"
ok 38 (unnamed assert)
ok 39 got peer2 "finish"
ok 40 (unnamed assert)
ok 41 got peer2 "end"
ok 42 (unnamed assert)
# duplex stream: send data one-way
ok 43 got peer1 "finish"
ok 44 (unnamed assert)
ok 45 got correct message
ok 46 got peer1 "end"
ok 47 (unnamed assert)
ok 48 got peer2 "finish"
ok 49 (unnamed assert)
ok 50 got peer2 "end"
ok 51 (unnamed assert)
# get config
# disable trickle
ok 52 only one `signal` event
ok 53 only one `signal` event
ok 54 peer1 is initiator
ok 55 peer2 is not initiator
ok 56 got correct message
ok 57 got correct message
ok 58 both peers closed
# disable trickle (only initiator)
ok 59 only one `signal` event for initiator
ok 60 at least one `signal` event for receiver
ok 61 peer1 is initiator
ok 62 peer2 is not initiator
ok 63 got correct message
ok 64 got correct message
ok 65 both peers closed
# disable trickle (only receiver)
ok 66 at least one `signal` event for initiator
ok 67 only one `signal` event for receiver
ok 68 peer1 is initiator
ok 69 peer2 is not initiator
ok 70 got correct message
ok 71 got correct message
ok 72 both peers closed

1..72
# tests 72
# pass  68
# fail  4
```

Sample Firefox failure:
```
unreachable code after return statement client.js:6610:4
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create client.js:1045:3
TAP version 13 client.js:11280:16
# get config client.js:11280:16
# detect WebRTC support client.js:11280:16
ok 1 builtin webrtc support client.js:11280:16
ok 2 should not be equal client.js:11280:16
# signal event gets emitted client.js:11280:16
WebRTC interfaces with the "moz" prefix (mozRTCPeerConnection, mozRTCSessionDescription, mozRTCIceCandidate) have been deprecated. test-bundle.js:11:5306
ok 3 got signal event client.js:11280:16
# data send/receive text client.js:11280:16
ok 4 (unnamed assert) client.js:11280:16
ok 5 (unnamed assert) client.js:11280:16
ok 6 peer1 is initiator client.js:11280:16
ok 7 peer2 is not initiator client.js:11280:16
not ok 8 should be equal client.js:11280:16
  --- client.js:11280:16
    operator: equal client.js:11280:16
    expected: '192.168.0.2' client.js:11280:16
    actual:   '10.100.1.6' client.js:11280:16
  ... client.js:11280:16
not ok 9 should be equal client.js:11280:16
  --- client.js:11280:16
    operator: equal client.js:11280:16
    expected: 49554 client.js:11280:16
    actual:   62468 client.js:11280:16
  ... client.js:11280:16
not ok 10 should be equal client.js:11280:16
  --- client.js:11280:16
    operator: equal client.js:11280:16
    expected: undefined client.js:11280:16
    actual:   '10.100.1.6' client.js:11280:16
  ... client.js:11280:16
not ok 11 should be equal client.js:11280:16
  --- client.js:11280:16
    operator: equal client.js:11280:16
    expected: undefined client.js:11280:16
    actual:   56551 client.js:11280:16
  ... client.js:11280:16
not ok 12 (unnamed assert) client.js:11280:16
  --- client.js:11280:16
    operator: ok client.js:11280:16
    expected: true client.js:11280:16
    actual:   false client.js:11280:16
  ... client.js:11280:16
peer1.remoteFamily is undefined test-bundle.js:9687:0
```

**Comment by paulkernfeld** [comment]
I just updated this; I thought it was wrtc only, but it happens in Firefox as well.

**Comment by paulkernfeld** [comment]
Here's a snippet with debug logs enabled. I think the issue is that there are multiple remote and local candidates coming in, so there's no guarantee that we are picking the right one in the [onStats function in index.js](https://github.com/feross/simple-peer/blob/master/index.js#L412).

```
  simple-peer [b3711b7] signal() +2ms
  simple-peer [b3711b7] signal() +1ms
  simple-peer [b3711b7] iceConnectionStateChange complete connected +105ms
  simple-peer [b3711b7] maybeReady pc true channel false +1ms
  simple-peer [null] iceConnectionStateChange complete connected +0ms
  simple-peer [null] maybeReady pc true channel false +0ms
  simple-peer [b3711b7] on channel open +1ms
  simple-peer [b3711b7] maybeReady pc true channel true +0ms
  simple-peer [b3711b7] connect remote: 192.168.0.2:61981 (IPv4) +3ms
  simple-peer [b3711b7] connect local: 192.168.0.2:59387 +0ms
  simple-peer [b3711b7] connect local: 10.100.1.6:59517 +0ms
  simple-peer [b3711b7] connect local: 10.100.1.6:52253 +0ms
  simple-peer [b3711b7] connect +1ms
  simple-peer [b3711b7] on channel open +0ms
  simple-peer [b3711b7] maybeReady pc true channel true +0ms
  simple-peer [b3711b7] connect remote: 192.168.0.2:59387 (IPv4) +2ms
  simple-peer [b3711b7] connect local: 192.168.0.2:61981 +0ms
  simple-peer [b3711b7] connect local: 10.100.1.6:59519 +0ms
  simple-peer [b3711b7] connect local: 10.100.1.6:64496 +0ms
  simple-peer [b3711b7] connect +0ms
ok 3 (unnamed assert)
ok 4 (unnamed assert)
ok 5 peer1 is initiator
ok 6 peer2 is not initiator
not ok 7 should be equal
  ---
    operator: equal
    expected: '192.168.0.2'
    actual:   '10.100.1.6'
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:58:7)
  ...
not ok 8 should be equal
  ---
    operator: equal
    expected: 59387
    actual:   52253
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:59:7)
  ...
not ok 9 should be equal
  ---
    operator: equal
    expected: '192.168.0.2'
    actual:   '10.100.1.6'
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:61:7)
  ...
not ok 10 should be equal
  ---
    operator: equal
    expected: 61981
    actual:   64496
    at: Peer.tryTest (/Users/paul/repos/simple-peer/test/basic.js:62:7)
  ...
ok 11 (unnamed assert)
ok 12 (unnamed assert)
ok 13 (unnamed assert)
ok 14 (unnamed assert)
  simple-peer [b3711b7] write: 9 bytes +5ms
  simple-peer [b3711b7] read: 9 bytes +1ms
ok 15 got correct message
```

**Comment by paulkernfeld** [comment]
BTW I'm interested in contributing for a fix, if there's a way that I can help.

**Comment by feross** [comment]
Figured out what is causing this! Working on a fix now.

**Comment by feross** [comment]
This is working a lot better, but still not perfect in Firefox. I relaxed the test a bit so we won't have any failures. Released in `simple-peer` 5.12.0.

**Comment by feross** [comment]
If you can figure out how to get `peer1.localAddress === peer2.remoteAddress` reliably in Firefox, that would be awesome!
